test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -71,36 +90,15 @@ call_log_manager::process_call_log_entry(json call_log_entry)
 
         switch (_api_map.at(method_name)) {
         case api_method::begin_transaction: {
-            const std::string session_id = call_log_entry["session_id"].get<std::string>();
-
-            /*
-             * We should not begin transactions on sessions with an ID that does not exist in the
-             * session map.
-             */
-            if (_session_map.find(session_id) == _session_map.end()) {
-                std::cerr << "Could not begin transaction, session does not exist (session ID: "
-                          << session_id << ")" << std::endl;
-                break;
-            }
-
-            /* Get the session from the session map. */
-            session_simulator *session = _session_map.at(session_id);
+            session_simulator *session = get_session(call_log_entry);
```

Comment:
You probably want to make sure it's not a nullptr here given the current implementation of the function. This comment is valid for the other places where you call that function.